### PR TITLE
2101/no jira fix to add statement defn

### PR DESCRIPTION
**Pull request by vdinovi**
I encountered this error when running `bundle exec rails generate declare_schema:migration` in web.
`/Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/declare_schema-0.5.0/lib/declare_schema/model/foreign_key_definition.rb:53:in `to_add_statement': wrong number of arguments (given 1, expected 0) (ArgumentError)`

It appears that [here](https://github.com/Invoca/declare_schema/blob/65f23e0c6d84c2a75f86668e83c3e6b4de57d1c3/lib/generators/declare_schema/migration/migrator.rb#L389) we call `DeclareSchema::Model::ForeignKeyDefinition#to_add_statement` with one argument however the definition doesn't support any. This method appears to parallel `DeclareSchema::Model::IndexDefinition#to_add_statement` which supports two arguments. So I've updated this method to be compatible by inserting two unused optional arguments into the former definition.

I also needed to add sqlite3 as a dev dependency to the Gemfile because creatiung the test app with `bundle exec rake test:prepare_testapp[force]` requires it.

```
Traceback (most recent call last):
	40: from bin/rails:6:in `<main>'
	39: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/bundler/gems/rails-439961ebda4e/activesupport/lib/active_support/dependencies.rb:274:in `require'
	38: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/bundler/gems/rails-439961ebda4e/activesupport/lib/active_support/dependencies.rb:240:in `load_dependency'
	37: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/bundler/gems/rails-439961ebda4e/activesupport/lib/active_support/dependencies.rb:274:in `block in require'
	36: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
	35: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
	34: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
	33: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
	32: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
	31: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/bundler/gems/rails-439961ebda4e/railties/lib/rails/commands.rb:17:in `<main>'
	30: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/bundler/gems/rails-439961ebda4e/railties/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
	29: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/bundler/gems/rails-439961ebda4e/railties/lib/rails/commands/commands_tasks.rb:50:in `generate'
	28: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/bundler/gems/rails-439961ebda4e/railties/lib/rails/commands/commands_tasks.rb:130:in `generate_or_destroy'
	27: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/bundler/gems/rails-439961ebda4e/railties/lib/rails/commands/commands_tasks.rb:123:in `require_command!'
	26: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/bundler/gems/rails-439961ebda4e/activesupport/lib/active_support/dependencies.rb:274:in `require'
	25: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/bundler/gems/rails-439961ebda4e/activesupport/lib/active_support/dependencies.rb:240:in `load_dependency'
	24: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/bundler/gems/rails-439961ebda4e/activesupport/lib/active_support/dependencies.rb:274:in `block in require'
	23: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
	22: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
	21: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
	20: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
	19: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
	18: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/bundler/gems/rails-439961ebda4e/railties/lib/rails/commands/generate.rb:13:in `<main>'
	17: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/bundler/gems/rails-439961ebda4e/railties/lib/rails/generators.rb:157:in `invoke'
	16: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	15: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/group.rb:232:in `dispatch'
	14: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:133:in `invoke_all'
	13: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:133:in `map'
	12: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:133:in `each'
	11: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:133:in `block in invoke_all'
	10: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	 9: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	 8: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/declare_schema-0.5.0/lib/generators/declare_schema/migration/migration_generator.rb:55:in `migrate'
	 7: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/declare_schema-0.5.0/lib/generators/declare_schema/migration/migrator.rb:298:in `generate'
	 6: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/declare_schema-0.5.0/lib/generators/declare_schema/migration/migrator.rb:298:in `map'
	 5: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/declare_schema-0.5.0/lib/generators/declare_schema/migration/migrator.rb:299:in `block in generate'
	 4: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/declare_schema-0.5.0/lib/generators/declare_schema/migration/migrator.rb:352:in `create_table'
	 3: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/declare_schema-0.5.0/lib/generators/declare_schema/migration/migrator.rb:382:in `create_constraints'
	 2: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/declare_schema-0.5.0/lib/generators/declare_schema/migration/migrator.rb:382:in `map'
	 1: from /Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/declare_schema-0.5.0/lib/generators/declare_schema/migration/migrator.rb:382:in `block in create_constraints'
/Users/vdinovi/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/declare_schema-0.5.0/lib/declare_schema/model/foreign_key_definition.rb:53:in `to_add_statement': wrong number of arguments (given 1, expected 0) (ArgumentError)
```

### CODE REVIEW

**Comment by ColinDKelley** [comment]
Crap, apparently that method has no tests? I want to fix that, but I can/ do that after you're unblocked.
And the sqlite3 dependency is fine. It wasn't needed if you always run tests as an appraisal. We can use the spiffy feature James added to appraisals where sqlite3 is in the base Gemfile but then removed from the `mysql` appraisals.

File: CHANGELOG.md [hunk 1]
```diff
@@ -92,6 +99,7 @@ using the appropriate Rails configuration attributes.
 ### Added
 - Initial version from https://github.com/Invoca/hobo_fields v4.1.0.
 
+[0.6.1]: https://github.com/Invoca/declare_schema/compare/v0.6.1...v0.6.2
```
Comment (ColinDKelley) [inline CHANGELOG.md:102]:
```suggestion
[0.6.2]: https://github.com/Invoca/declare_schema/compare/v0.6.1...v0.6.2
```

File: Gemfile [hunk 2]
```diff
@@ -19,3 +19,4 @@ gem 'responders'
 gem 'rspec'
 gem 'rubocop'
 gem 'yard'
+gem 'sqlite3',           '~>1.4.0'
```
Comment (ColinDKelley) [inline Gemfile:22]: For `~>`, we always want exactly one decimal place, so it works as SemVer-compatible.
```suggestion
gem 'sqlite3',           '~> 1.4'
```

**Review by ColinDKelley — APPROVED** [review]
Thanks for the PR. Just a couple suggestions here.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,13 @@ Inspired by [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 Note: this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.6.2] - 2021-01-06
+### Changed
+- added compatibility params to DeclareSchema::Model::ForeignKeyDefinition#to_add_statement
```
Comment (jebentier) [inline CHANGELOG.md:9]:
```suggestion
### Fixed
- Fixed a but in migration generation caused by `DeclareSchema::Model::ForeignKeyDefinition#to_add_statement`
not being passed proper arguments
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,13 @@ Inspired by [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
...
+
+### Added
+- sqlite3 as dev dependency
```
Comment (jebentier) [inline CHANGELOG.md:12]:
```suggestion
### Added
- Added `sqlite3` as dev dependency for local development
```

**Review by jebentier — APPROVED** [review]
Looks good, just would like a couple changes to the changelog to follow conventions and be more helpful

**Comment by vdinovi** [comment]
Sorry for the delay. All comments have been addressed. I've QAd this by testing out the changes using a local web branch. Plan to merge once I get a green build.

**Pull request merged**